packages/babel-helpers/src/helpers.js

```diff
@@ -372,6 +372,29 @@ helpers.extends = defineHelper(`
   }
 `);
 
+helpers.spreadByDefine = defineHelper(`
+  export default function _spreadByDefine(target) {
+    for (var i = 1; i < arguments.length; i++) {
+      var source = (arguments[i] != null) ? arguments[i] : {};
+      var ownKeys = Object.keys(source);
+      if (typeof Object.getOwnPropertySymbols === 'function') {
+        ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
+          return Object.getOwnPropertyDescriptor(source, sym).enumerable;
+        }));
+      }
+      ownKeys.forEach(function(key) {
```

Comment:
for better perf you could use existing `defineProperty` helper here, which does an additional check before using `Object.defineProperty`

it should be possible to add `import defineProperty from "defineProperty"` to use it in the created helper

Reply:
Mhh ok cool! Thanks for the tip, I'll try that

Reply:
Just checked that helper's code and what it does is just:

```js
    if (key in obj) {
      Object.defineProperty(obj, key, {
        value: value,
        enumerable: true,
        configurable: true,
        writable: true
      });
    } else {
      obj[key] = value;
    }
```

So seems that in this context it won't be helpful at the end... cause it's the intended behavior to always define a property (and not to do an assignment)... 😌

Reply:
how is the outcome different? from what I understand the point of using `Object.defineProperty` is to omit potential setters, but if there is no property with the given key there is no setter for this key and a simple assignment can be used to cause the same effect (default descriptor of assignment is `{ enumerable: true, configurable: true, writable: true }`)

Reply:
Oh, I think you're right! 😅  I was missing that important detail about the `in` check...

So, how in practice I could use an helper inside another helper? You mentioned the `import defineProperty from "defineProperty"` but I can't see any practical example... I mean, everything is declared as string like:
```
defineHelper(`…source code…`);
```

:thinking:

Reply:
example of adding a helper dependency - [here](https://github.com/nuragic/babel/blob/00ae9810bb45a543eca5543cc54aec145a6bc141/packages/babel-helpers/src/helpers.js#L97)

Reply:
Thanks! Fixed